### DISCUSSION TITLE
TTL mechanism for partitions

### DISCUSSION THREAD

**Discussion by rkozlo** [ClickHouse/ClickHouse#28185] [General]
Hi,
i have huge database with huge tables and i need to tier it on hot and cold storage. What i found is of course TTL which is good but i would prefer something different. Is there something like TTL for partitions? I would like to move the whole partition on other volume what i think will be more efficient in my case. I know i can move partition with ALTER TABLE but i look for some automatic solution.

**Comment by SaltTan** [comment]
You can use TTL to move parts.
If your TTL expression aligns with the partitioning expression, the whole partition will be moved at once.

**Reply by rkozlo** [reply]
What about situation when it is partitioned by month? Partition will be splited or be moved based on max_date col?
